src/core/instance/state.js

```diff
@@ -60,6 +60,17 @@ const isReservedProp = {
   slot: 1
 }
 
+function parseObjectOption (vm: Component, option: Object, name: string) {
+  if (!isPlainObject(option)) {
+    process.env.NODE_ENV !== 'production' && warn(
```

Comment:
it may be good to provide a fallback (empty object) if a wrong type is used in prod mode